### PR TITLE
Add eslint rules for missing or improper jsdoc tags

### PR DESCRIPTION
**Pull request by budnix**
### Context

Adds jsdoc tags checking to the linting process.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6314

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by budnix** [comment]
@wojciechczerniak It's worth to mention that the PR implements the recommended rules which by default warn the developer when something is wrong with `jsdoc` tags. Hence, the CI shows a green light even when the docs are incomplete. Maybe it's better to report this as an error?

File: src/plugins/undoRedo/undoRedo.js [hunk 1]
```diff
@@ -267,6 +271,8 @@ UndoRedo.Action.prototype.redo = function() {};
  * Change action.
  *
  * @private
+ * @param changes
```
Comment (pnowak) [inline src/plugins/undoRedo/undoRedo.js]: I'm curious why in this file almost all @params are without type and descriptions?

File: src/plugins/exportFile/types/_base.js [hunk 2]
```diff
@@ -46,7 +46,8 @@ class BaseType {
   /**
    * Merge options provided by users with defaults.
    *
-   * @return {Object} Returns new options object.
+   * @returns {object} Returns new options object.
```
Comment (pnowak) [inline src/plugins/exportFile/types/_base.js]: Swap places @returns and @param, please.

File: src/plugins/filters/component/condition.js [hunk 3]
```diff
@@ -98,10 +98,11 @@ class ConditionComponent extends BaseComponent {
...
    * Update state of component.
-   * @param {Object} condition Object with keys:
-   *  * `command` Object, Command object with condition name as `key` property.
-   *  * `args` Array, Condition arguments.
-   * @param column Physical column index.
+   *
+   * @param {object} condition The condition object.
+   * @param {object} condition.command The command object with condition name as `key` property.
+   * @param {Array} condition.args An array of values to compare.
+   * @param {number} column Physical column index.
```
Comment (pnowak) [inline src/plugins/filters/component/condition.js]: But function has 2 arguments.

File: src/helpers/browser.js [hunk 4]
```diff
@@ -22,40 +22,67 @@ const browsers = {
   safari: tester((ua, vendor) => /Safari/.test(ua) && /Apple Computer/.test(vendor)),
 };
 
+/**
+ *
```
[the remark this answers is not in the document]
Comment (budnix) [reply src/helpers/browser.js]: Those type function parameters (with destructing) are not validated by jsdoc plugin (https://github.com/gajus/eslint-plugin-jsdoc#deconstructing-function-parameter). Good catch!

File: src/plugins/undoRedo/undoRedo.js (hunk 1 above)
Comment (budnix) [reply src/plugins/undoRedo/undoRedo.js]: Resolved within https://github.com/handsontable/handsontable/pull/6647/commits/55ab30a38a6b414bb1204617b1989f89169664c8.

File: src/plugins/exportFile/types/_base.js (hunk 2 above)
Comment (budnix) [reply src/plugins/exportFile/types/_base.js]: Resolved within https://github.com/handsontable/handsontable/commit/902cb8ba8578dcc38dc0d342bd222a703c32a68a.

File: src/plugins/filters/component/condition.js (hunk 3 above)
Comment (budnix) [reply src/plugins/filters/component/condition.js]: This actually is a correct description. If the function argument is passed as an object we can document what properties that object contains. Here the docs say that passed the `condition` object must have `command` and `args` keys. Moreover, it is generated correctly https://handsontable.com/docs/7.3.0/ConditionComponent.html#updateState.

File: src/helpers/browser.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (budnix) [reply src/helpers/browser.js]: Resolved within https://github.com/handsontable/handsontable/commit/8930603433695fef4395293fd6508870dc56efd3.

File: src/helpers/object.js [hunk 5]
```diff
@@ -189,26 +190,33 @@ export function mixin(Base, ...mixins) {
...
   return JSON.stringify(object1) === JSON.stringify(object2);
 }
 
 /**
  * Determines whether given object is a plain Object.
- * Note: String and Array are not plain Objects
- * @param {*} obj
+ * Note: String and Array are not plain Objects.
+ *
+ * @param {*} object An object to ckeck.
```
Comment (pnowak) [inline src/helpers/object.js]: check ;)

Comment (budnix) [reply src/helpers/object.js]: Fixed within https://github.com/handsontable/handsontable/pull/6647/commits/0030d14603a0066b58475b14487689e3bbe643ab 👍

**Pull request merged**